packages/utils/src/get-snark-artifacts/config.ts

```diff
@@ -0,0 +1,87 @@
+import { Artifact, Proof, SnarkArtifacts, Version } from "../types"
+
+const ARTIFACTS_BASE_URL = "https://unpkg.com/@zk-kit"
+
+const getPackageVersions = async (proof: Proof) =>
+    fetch(`${ARTIFACTS_BASE_URL}/${proof}-artifacts`)
+        .then((res) => res.json())
+        .then((data) => Object.keys(data.versions))
+
+export async function GetSnarkArtifactUrls({
+    proof,
+    version
+}: {
+    proof: Proof.EDDSA
+    version?: Version
+}): Promise<SnarkArtifacts>
+export async function GetSnarkArtifactUrls({
+    proof,
+    numberOfInputs,
+    version
+}: {
+    proof: Proof.POSEIDON
+    numberOfInputs: number
+    version?: Version
+}): Promise<SnarkArtifacts>
+export async function GetSnarkArtifactUrls({
+    proof,
+    treeDepth,
+    version
+}: {
+    proof: Proof.SEMAPHORE
+    treeDepth: number
+    version?: Version
+}): Promise<SnarkArtifacts>
+export async function GetSnarkArtifactUrls({
+    proof,
+    numberOfInputs,
+    treeDepth,
+    version
+}: {
+    proof: Proof
+    numberOfInputs?: number
+    treeDepth?: number
+    version?: Version
+}) {
+    if (version !== undefined) {
+        const availableVersions = await getPackageVersions(proof)
+
+        if (!availableVersions.includes(version))
+            throw new Error(
+                `Version ${version} is not available for ${proof} proofs, available versions are: ${availableVersions}`
+            )
+    } else {
+        version ??= "latest"
+    }
+
+    const BASE_URL = `https://unpkg.com/@zk-kit/${proof}-artifacts@${version}`
+
+    switch (proof) {
+        case Proof.EDDSA:
+            return {
+                [Artifact.WASM]: `${BASE_URL}/${proof}.${Artifact.WASM}`,
+                [Artifact.ZKEY]: `${BASE_URL}/${proof}.${Artifact.ZKEY}`
+            }
+
+        case Proof.POSEIDON:
+            if (numberOfInputs === undefined) throw new Error("numberOfInputs is required for Poseidon proof")
```

Comment:
`@zk-kit/utils` provides some utility functions to check if parameters are defined or have a specific type. We could consider using them here.

Reply:
https://www.typescriptlang.org/play?#code/KYDwDg9gTgLgBAMwK4DsDGMCWEV0wZwBFgFMVgATACgDcBDAGyWAC446UBPASjYCMIEBsA5wA3gCg40uFGAwkUXDE5hgEBHHpNgcAIQBeA3ABEqCiTKUTEgL4SJoSLESoM2XHICOSTHOKk5NRgdFB0ALbywFAAaozMbBycADRwIWGRMNEAchGscPgwUGQA5tziUjKYmlR6BAFWwaF5WbHxwNzlkjI9cDAAFlAQAO5w5KMAKqrAAKJQQ1BUAAYACs2Z0XAA5AAkYuktOXm2W3j4YxDwFoGUS9yV0vb2EsjoWDiIglR8oQD8LCgkOE+NEug9ZMAfH5gA0gt9QqkTD8oCZ7j1qnB4VA4AAeOAARnKAyGo3GcDmCyoJnCSEKcBBcAAfATUXYJEA

Unfortunately although `requireDefined` ensures the value is defined, it appears that in this case TS isn't smart enough and can't statically analyze the code flow to understand that `numberOfInputs` can't be defined, and it errors with `'numberOfInputs' is possibly 'undefined'`. So I'll keep the explicit check.